server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3348,6 +3328,25 @@ protected void updateDiskOfferingTagsIfIsNotNull(String tags, DiskOfferingVO dis
         }
     }
 
+    /**
+     * Check the host tags parameters to the service offering
+     * <ul>
+     *     <li>If host tags is null, do nothing and return.</li>
+     *     <li>If host tags is not null, set host tag to the service offering.</li>
+     *     <li>If host tags is an blank string, set null on service offering tag.</li>
+     * </ul>
+     */
+    protected void updateServiceOfferingHostTagsIfNotNull(String hostTags, ServiceOfferingVO offering) {
+        if (hostTags == null) {
+            return;
+        }
+        if (StringUtils.isNotBlank(hostTags)) {
+            offering.setHostTag(hostTags);
+        } else {
+            offering.setHostTag(null);
+        }
+    }
```

Comment:
given this is identical to `updateOfferingTagsIfIsNotNull` in logic, I'd sugest creating a `TaggedVO` interface and unify the methods!?

Reply:
thanks, @DaanHoogland, for the review! I'll think about unifying the method.